p2p/peer.go

```diff
@@ -43,6 +44,9 @@ type Peer interface {
 	Set(key string, value any)
 	Get(key string) any
 
+	SetHeight(h int64)
```

Comment:
These two methods overlap with `Get` and `Set` but I couldn't used them in `execution.go` without introducing a circular dependency with consensus. In mempool and evidence we call `peer.Get(types.PeerStateKey).(*PeerState)` to obtain the height. In the future we could replace this call by `peer.Height()`.